tests/integration/models/os.spec.ts

```diff
@@ -664,6 +665,33 @@ describe('OS model', function () {
 					});
 			});
 
+			it('should be able to configure v2 image with a provisioning key name', async function () {
+				const configOptions = {
+					appUpdatePollInterval: 72,
+					network: 'wifi' as const,
+					wifiKey: 'foobar',
+					wifiSsid: 'foobarbaz',
+					ip: '1.2.3.4',
+					gateway: '5.6.7.8',
+					netmask: '9.10.11.12',
+					version: '2.11.8+rev1.prod',
+					provisioningKeyName: 'provision-key',
+				};
+
+				await balena.models.os.getConfig(ctx.application.id, configOptions);
+
+				const provisioningKeys =
+					await balena.models.apiKey.getProvisioningApiKeysByApplication(
+						ctx.application.id,
+					);
+
+				expect(provisioningKeys).to.be.an('array');
+				expect(provisioningKeys.length).is.not.equal(0);
+				expect(
+					_.findIndex(provisioningKeys, ['name', 'provision-key']),
+				).to.not.equal(-1);
```

Comment:
1. get the provisioning api keys before your call
2. generate the named config
3. fetch the provisioning api keys ordered by created_at desc
4. filter the newly retrieved api keys to exclude the ones that are in the list of IDs of the api keys of step 1
5. expect that what's left is has length of 1
6. expect hat the single new apikey has a property name that equals the name you provided.

Reply:
Okay - so on very similar lines like the last time (https://github.com/balena-io/balena-sdk/pull/1107/files#diff-6e5edef2fac65498ec08862f218a130979112a22de7b03b6405eda18c16eb445R692).